The way config is handled has been overhauled

tuupke opened the pull request:
Everything related to config has now been placed in a single command.
All interaction with the config file must take place through this
command. It can be invoked with `contest config` and provides, list,
delete, set, unset, login, and logout sub-commands.

nickygerritsen commented on README.md:
```diff
@@ -1,108 +1,141 @@
+| ------- | ------- |
+| `contest set url <url>` | Specify the base contest URL. Can also be specified via -url. |
+| `contest set id <id>` | Only required in cases where there is more than one contest *and* there isn't an obvious default contest. Can also be specified via using -c. |
+| `contest login <user> <password>` | Loaded from .netrc if not specified. Not required when using IP/host auto-login. Can also be specified via -u and -p. |
```
Not netrc I’d say but $HOME/.icpc/config.yaml

tuupke committed:
The way config is handled has been overhauled

Everything related to config has now been placed in a single command.
All interaction with the config file must take place through this
command. It can be invoked with `contest config` and provides, list,
delete, unset, login, and logout subcommands.

deboer-tim commented:
I like the cli pattern better with set/list/delete, but I think the only config we're likely to have is the current short & fixed list (i.e. url/id/insecure/user/password) and maybe we'll need one or two more in the future. Right now if I run "./contest set" it shows me that only url and id can be set - doesn't this lose that discoverability and allow any property to be set? e.g. I set uri or baseUrl, why isn't it working?

tuupke commented:
The properties that can be set can be easily added to the usage string. Will apply this change somewhere today

deboer-tim commented:
Was hoping we can both document and limit what you can set, i.e. "contest config passwrd" should fail.

Ideally I'd like to verify config in the future as well (e.g. when you enter credentials it'd tell you if they work) but I think we can add that in as a general validation.
